modes/vi-mode/core.lisp

```diff
@@ -55,27 +59,50 @@
   (setf (element-name *modeline-element*) name))
 
 
-(defstruct vi-state
-  name
-  keymap
-  post-command-hook
-  enable-hook
-  disable-hook
-  cursor-color)
+(defclass vi-state ()
+  ((tag
+  :initarg :tag
+  :reader tag)
```

Comment:
What is a tag?

Reply:
I remove this, this is short hand for the current state. Not needed now.